src/sql/src/plan/query.rs

```diff
@@ -2692,7 +2748,43 @@ impl<'a> QueryContext<'a> {
                 .chain(std::iter::once(relation_type))
                 .cloned()
                 .collect(),
+            ctes,
+        }
+    }
+
+    /// Resolves `name` to a table expr, i.e. getting a catalog table or
+    /// inlining a CTE.
+    pub fn resolve_table_expression(
+        &self,
+        name: ObjectName,
+    ) -> Result<(PartialName, RelationExpr, RelationDesc), PlanError> {
```

Comment:
I think this method breaks out a bit more cleanly if you return `(RelationExpr, Scope)`? Then you don't need to return the `PartialName` and `RelationDesc` separately, since those get packaged up into to scope. (I think the call to `plan_table_alias` still belongs in the caller though.)

Also, I think it should be called `resolve_table_name` for consistency with the other similar methods! (They're named after what they resolve from, not what they resolve to.)